### PR TITLE
Fix build in fix-r-num-abs-evaluation branch

### PR DESCRIPTION
**Pull request by lzutao**
This fix build in pull request radare/radare2#12072.

### CODE REVIEW

**Comment by lzutao** [comment]
Closed to send pull request to same branch fix-r-num-abs-evaluation on radare2.

File: old/t/feat_grep [hunk 1]
```diff
@@ -179,7 +179,7 @@ block    0x100
 fd       42
 block    0x100
 size     0x400
-humansz  1K
+humansz  1.0K
```
Comment (radare) [inline old/t/feat_grep]: i prefer the current behaviour

File: unit/test_unum.c [hunk 2]
```diff
@@ -4,14 +4,44 @@
...
-	mu_assert_streq (r_num_units ((char *)buf, 1200 * 1024 * 1024), "1.2G", "G");
+	char humansz[8];
+	const struct {
+		const char *expected_res;
+		const char *message;
+		ut64 num;
+	} test_cases[] = {
+		{ "0",        "B", 0ULL },
+		{ "512",      "B", 512ULL },
+		{ "1.0K",     "K", 1ULL << 10 },
```
Comment (radare) [inline unit/test_unum.c]: same here. Better 1K than 1.0K or 1.0M

File: old/t/cmd_help [hunk 3]
```diff
@@ -290,7 +290,7 @@ ARGS=
 CMDS="? -1"
 EXPECT='hex     0xffffffffffffffff
 octal   01777777777777777777777
-unit    17179869184.0G
+unit    16.0E
```
Comment (radare) [inline old/t/cmd_help]: nice one

File: unit/test_unum.c (hunk 2 above)
Comment (lzutao) [reply unit/test_unum.c]: for only multiple of 1024 or just round 1.0 to 1

**Comment by lzutao** [comment]
@radare You want 1.0 for only multiple of 1024 or just round 1.0 to 1.

**Comment by radare** [comment]
just check if the mod is 0

File: unit/test_unum.c (hunk 2 above)
Comment (radare) [reply unit/test_unum.c]: just remove if mod is 0

Comment (radare) [reply unit/test_unum.c]: like it was before

Comment (lzutao) [reply unit/test_unum.c]: There are two ways to do this but each would make a little different
in result:

* The first way is like the original code:

  We check if `fnum` will have a zeo right after decimal point. If it is,
  we output with `'%.0f'` only, which remove the fractional part.

* The second way is to check if `fnum` is indeed integer

  We check that condition by using `ceill` function. This way only checks
  if `fnum` is integer, so if input is `num=1025`, we will have `1.0K` instead
  of `1K` in the first way.

So what would you choose, @radare?

**Pull request merged**